contracts/proposers/BurnAuction.sol

```diff
@@ -0,0 +1,101 @@
+pragma solidity ^0.5.15;
+import { Chooser } from "./Chooser.sol";
+import { Bitmap } from "../libs/Bitmap.sol";
+import { Ownable } from "@openzeppelin/contracts/ownership/Ownable.sol";
+
+contract BurnAuction is Chooser, Ownable {
+    uint32 constant BLOCKS_PER_SLOT = 100;
+    uint32 constant DELTA_BLOCKS_INITIAL_SLOT = 1000;
+
+    // First block where the first slot begins
+    uint256 public genesisBlock;
+
+    // Bid is a structure indicating that
+    // "operator" is willing to burn certain
+    // "amount" of ether in order to forge a slot.
+    // "initialized" is used to indicate that the bid is valid (used to differenciate an empty bid)
+    struct Bid {
+        address payable operator;
+        uint128 amount;
+        bool initialized;
+    }
+
+    // auction is a relation of the best bid for each slot
+    mapping(uint32 => Bid) public auction;
+    // slot -> hasProposed
+    mapping(uint256 => uint256) public hasProposed;
```

Comment:
Diff to iden3: We track if a winner has proposed in the Burn auction contract, while the iden3 track in the rollup contract.

Reply:
Curious to know why is it needed to track if a winner hasProposed or not.

Reply:
If we only allow a winner to propose 1 batch per slot then we need this check.

Reply:
[winner can propose as many batches they want] so no need to track hasProposed now